sdk/trace.go

```diff
@@ -54,14 +56,17 @@ var _ trace.Tracer = tracer{}
 
 func (t tracer) Start(ctx context.Context, name string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
 	var psc trace.SpanContext
-	span := &span{sampled: true}
+	sampled := true
+	span := new(span)
 
 	// Ask eBPF for sampling decision and span context info.
-	t.start(ctx, span, &psc, &span.sampled, &span.spanContext)
+	t.start(ctx, span, &psc, &sampled, &span.spanContext)
+
+	span.sampled.Store(sampled)
 
 	ctx = trace.ContextWithSpan(ctx, span)
 
-	if span.sampled {
+	if sampled {
```

Comment:
will `sampled` always be true here? I didn't see a test for `span.traces` and `span.span` that are set in this block but maybe I missed it

Reply:
If the eBPF program sets it to false based on its sampler `sampled` will be `false`.

Reply:
ahh right, this is the magic variable